### PR TITLE
Extract settlors forms

### PR DESCRIPTION
**Pull request by JoshuaH01**

### CODE REVIEW

File: app/controllers/deceased_settlor/SettlorDateOfDeathYesNoController.scala [hunk 1]
```diff
@@ -38,13 +38,13 @@ class SettlorDateOfDeathYesNoController @Inject()(
...
                                          requireData: DataRequiredAction,
-                                         formProvider: SettlorDateOfDeathYesNoFormProvider,
+                                         yesNoFormProvider: YesNoFormProvider,
                                          requiredAnswer: RequiredAnswerActionProvider,
                                          val controllerComponents: MessagesControllerComponents,
                                          view: SettlorDateOfDeathYesNoView
                                  )(implicit ec: ExecutionContext) extends FrontendBaseController with I18nSupport {
 
-  val form: Form[Boolean] = formProvider()
+  val form: Form[Boolean] = yesNoFormProvider.withPrefix("settlorDateOfDeath")
```
Comment (jassalrichy) [inline app/controllers/deceased_settlor/SettlorDateOfDeathYesNoController.scala]: Should prefix not  be "settlorDateOfDeathYesNo"?

File: test/controllers/deceased_settlor/SettlorDateOfDeathYesNoControllerSpec.scala [hunk 2]
```diff
@@ -26,8 +26,7 @@ import views.html.deceased_settlor.SettlorDateOfDeathYesNoView
 
 class SettlorDateOfDeathYesNoControllerSpec extends SpecBase {
 
-  val formProvider = new SettlorDateOfDeathYesNoFormProvider()
-  val form = formProvider()
+  val form = new YesNoFormProvider().withPrefix("settlorDateOfDeath")
```
Comment (jassalrichy) [inline test/controllers/deceased_settlor/SettlorDateOfDeathYesNoControllerSpec.scala]: Should prefix not  be "settlorDateOfDeathYesNo"?

File: test/views/deceased_settlor/SettlorsNINoYesNoViewSpec.scala [hunk 3]
```diff
@@ -27,7 +27,7 @@ class SettlorsNINoYesNoViewSpec extends YesNoViewBehaviours {
 
   val messageKeyPrefix = "settlorsNINoYesNo"
 
-  val form = new SettlorsNINoYesNoFormProvider()()
+  val form = new YesNoFormProvider().withPrefix("settlorsNINoYesNo")
```
Comment (jassalrichy) [inline test/views/deceased_settlor/SettlorsNINoYesNoViewSpec.scala]: use messageKeyPrefix

File: app/controllers/deceased_settlor/SettlorDateOfDeathYesNoController.scala (hunk 1 above)
Comment (JoshuaH01) [reply app/controllers/deceased_settlor/SettlorDateOfDeathYesNoController.scala]: Yes! Thank you. I have changed that

File: test/controllers/deceased_settlor/SettlorDateOfDeathYesNoControllerSpec.scala (hunk 2 above)
Comment (JoshuaH01) [reply test/controllers/deceased_settlor/SettlorDateOfDeathYesNoControllerSpec.scala]: Complete

File: test/views/deceased_settlor/SettlorsNINoYesNoViewSpec.scala (hunk 3 above)
Comment (JoshuaH01) [reply test/views/deceased_settlor/SettlorsNINoYesNoViewSpec.scala]: complete

**Pull request merged**